packages/dhis2/src/Adaptor.js

```diff
@@ -229,20 +229,45 @@ export function create(resourceType, data, options = {}, callback = false) {
   return state => {
     console.log(`Preparing create operation...`);
 
+    let mappedOptions;
+
+    if (shouldUseNewTracker(resolvedResourceType)) {
```

Comment:
We need to do resolvedOptions before we create `mappedOptions`. Otherwise, if `options` is passed as a function, things will blow up.

If you're to clear on how and why we resolve options, let me know and I can walk you through it